### PR TITLE
Add new language ST

### PR DESCRIPTION
**Pull request by Serhioromano**
A new PR for the same ST language discussed in #1735. I hope this one will pass all the tests. Changed the name of the language from `st` to `iecst` to avoid conflicts with `Smalltalk`.

### CODE REVIEW

**Comment by Serhioromano** [comment]
Here is the error.



What does that means it have to be shell? And where?

**Comment by gusbemacbe** [comment]
I believe you missed the semicolon in the `return`:

```javascript
function(hljs) {
    return {
 # ...
    }
} 
```

Instead, like:

```javascript
function(hljs) {
    return {
 # ...
    };
} 
```
The most off files used `;` in the `return`.

**Comment by Serhioromano** [comment]
@gusbemacbe Nope. Not this one. Still fail checks with absolutely the same error.

**Comment by gusbemacbe** [comment]
Sorry, I went to bed. I reviewed again your codes and there was no error. Strange, maybe because of the alia as @ldct reported at his #1779.

**Comment by Serhioromano** [comment]
@gusbemacbe nope. Deleted same alias but same error in Travis. Expected to be shell.

**Comment by gusbemacbe** [comment]
@Serhioromano, I am going to call @isagalaev, @ldct, @marcoscaceres and @SirCmpwn to analyse this bug now.

**Comment by marcoscaceres** [comment]
I was unable to track down the cause, unfortunately. 

@gusbemacbe, with your code removed, does it error? Or is it only when your new code is added?

**Comment by gusbemacbe** [comment]
@marcoscaceres, unfortunately, no, I believe it be a Mocha bug.

**Comment by marcoscaceres** [comment]
@gusbemacbe, I guess the next thing to try is updating to latest Mocha. Could you please try that? 

To be sure, I got a memory leak error from Bluebird, but we can ignore that if you also see it.

**Comment by gusbemacbe** [comment]
@marcoscaceres 

You have just open the new issue, suggesting us to drop Bluebird and use utils.promisify, but do I need to reinstall to remove BlueBird and update to Mocha, applying utils.promisify? I am confused.

**Comment by marcoscaceres** [comment]
> You have just open the new issue, suggesting us to drop Bluebird and use utils.promisify, but do I need to reinstall to remove BlueBird and update to Mocha, applying utils.promisify? I am confused.

Sorry about the confusion. No, you don't need to do any of the above. 

Just do:

```JS
npm install mocha@latest
npm run test
```

If you see:

```Bash
Error: global leak detected: document
```

That's ok. But if your error above persists, we will know it's something do to with your PR (which is fine, we can fix it 💪!).

**Comment by gusbemacbe** [comment]
It seems that after installing Highlight.js with `npm`, it still uses the old version, i.e., 9.12, maybe can the old version be the cause of these errors?

**Comment by marcoscaceres** [comment]
@gusbemacbe, perhaps, but let's make sure we exhaust the current investigation before going down that route. I gather from your response that what I suggested still failed with the same error you were getting as before?

**Comment by gusbemacbe** [comment]
Sorry for my late. I had a lot of errors with npm. I had to make a clean install of npm. But I have still Mocha errors, but the errors are different:

```shell
npm run test

> gusbemacbe@1.0.0 test /home/gusbemacbe
> mocha

Warning: Could not find any test files matching pattern: test
No test files found
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gusbemacbe@1.0.0 test: `mocha`
npm ERR! Exit status 1
npm ERR! 
```

**Comment by marcoscaceres** [comment]
@gusbemacbe, by your output, looks like you are not in the highlight.js directory. Can you check?

**Comment by gusbemacbe** [comment]
Unfortunatelly with Highlight.js and Mocha installed, the same errors still insists.

```shell
~$ cd node_modules
~/node_modules$ ls
accepts              depd                  growl              mime-types        qs
array-flatten        destroy               has-flag           minimatch         range-parser
balanced-match       diff                  he                 minimist          raw-body
body-parser          ee-first              highlight.js       mkdirp            safe-buffer
brace-expansion      encodeurl             http-errors        mocha             send
browser-stdout       escape-html           iconv-lite         ms                serve-static
bytes                escape-string-regexp  inflight           negotiator        setprototypeof
commander            etag                  inherits           npm               statuses
concat-map           express               ipaddr.js          once              supports-color
content-disposition  finalhandler          media-typer        on-finished       type-is
content-type         forwarded             merge-descriptors  parseurl          unpipe
cookie               fresh                 methods            path-is-absolute  utils-merge
cookie-signature     fs.realpath           mime               path-to-regexp    vary
debug                glob                  mime-db            proxy-addr        wrappy
~/node_modules$ npm run test /highlight.js

> gusbemacbe@1.0.0 test /home/gusbemacbe
> mocha "/highlight.js"

Warning: Could not find any test files matching pattern: /highlight.js
No test files found
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gusbemacbe@1.0.0 test: `mocha "/highlight.js"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the gusbemacbe@1.0.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

~/node_modules$ cd highlight.js
~/node_modules/highlight.js$ npm run test

> highlight.js@9.12.0 test /home/gusbemacbe/node_modules/highlight.js
> mocha test/

Warning: Could not find any test files matching pattern: test/
No test files found
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! highlight.js@9.12.0 test: `mocha test/`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the highlight.js@9.12.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
```

**Comment by marcoscaceres** [comment]
@gusbemacbe, ok, I see what's happened. You are testing the wrong thing :) You should never need to go into the `node_modules` directory - that's the distribution build, which won't match the repository. 

I'm going to try something different. Don't worry.

**Comment by Serhioromano** [comment]
Any news? It takes so long.

File: src/languages/iecst.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            },
+            {
+                className: 'symbol',
+                begin: '\%(I|Q|M)(X|B|W|D|L)[0-9\.]*',
+                relevance: 0
+            },
+            {
+                className: 'symbol',
+                begin: '\%(I|Q|M)[0-9\.]*',
+                relevance: 0
```
Comment (pascalfree) [inline src/languages/iecst.js]:
```suggestion
```

File: src/languages/iecst.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            },
+            {
+                className: 'symbol',
+                begin: '\%(I|Q|M)(X|B|W|D|L)[0-9\.]*',
+                relevance: 0
+            },
+            {
+                className: 'symbol',
+                begin: '\%(I|Q|M)[0-9\.]*',
```
Comment (pascalfree) [inline src/languages/iecst.js]:
```suggestion
                begin: '\%(I|Q|M)[0-9\.]*'
```

File: src/languages/iecst.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            },
+            {
+                className: 'number',
+                begin: '[a-zA-Z_]*#[a-zA-Z_]*',
+                relevance: 0
+            },
+            {
+                className: 'symbol',
+                begin: '\%(I|Q|M)(X|B|W|D|L)[0-9\.]*',
+                relevance: 0
```
Comment (pascalfree) [inline src/languages/iecst.js]:
```suggestion
```

File: src/languages/iecst.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            },
+            {
+                className: 'number',
+                begin: '[a-zA-Z_]*#[a-zA-Z_]*',
+                relevance: 0
+            },
+            {
+                className: 'symbol',
+                begin: '\%(I|Q|M)(X|B|W|D|L)[0-9\.]*',
```
Comment (pascalfree) [inline src/languages/iecst.js]:
```suggestion
                begin: '\%(I|Q|M)(X|B|W|D|L)[0-9\.]*'
```

File: src/languages/iecst.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            },
+            {
+                className: 'string',
+                begin: '"', end: '"',
+                contains: [hljs.BACKSLASH_ESCAPE, { begin: '""' }]
+            },
+            {
+                className: 'symbol',
+                begin: '(T|DT|TOD)#[0-9:-_shmyd]*',
+                relevance: 0
```
Comment (pascalfree) [inline src/languages/iecst.js]:
```suggestion
```

File: src/languages/iecst.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                contains: [hljs.BACKSLASH_ESCAPE, { begin: '\'\'' }]
+            },
+            {
+                className: 'string',
+                begin: '"', end: '"',
+                contains: [hljs.BACKSLASH_ESCAPE, { begin: '""' }]
+            },
+            {
+                className: 'symbol',
+                begin: '(T|DT|TOD)#[0-9:-_shmyd]*',
```
Comment (pascalfree) [inline src/languages/iecst.js]:
```suggestion
                begin: '(T|DT|TOD)#[0-9:-_shmyd]*'
```

File: src/languages/iecst.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        contains: [
+            {
+                className: 'string',
+                begin: '\'', end: '\'',
+                contains: [hljs.BACKSLASH_ESCAPE, { begin: '\'\'' }]
+            },
+            {
+                className: 'string',
+                begin: '"', end: '"',
+                contains: [hljs.BACKSLASH_ESCAPE, { begin: '""' }]
```
Comment (pascalfree) [inline src/languages/iecst.js]:
```suggestion
                contains: [hljs.BACKSLASH_ESCAPE, { begin: '""' }],
                relevance: 0
```

File: src/languages/iecst.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            'var var_global end_var var_input var_out var_output var_in_out var_temp var_interval var_access var_config',
+            literal: 'false true null ',
+            built_in: 'array pointer int sint dint lint usint uint udint ulint real lreal time date time_of_day date_and_time dt tod string bool byte world dworld lworld ref_to any_num any_int any_string',
+            function: 'mod abs acos asin atan cos exp expt ln log sin sqrt tan sel max min limit mux shl shr rol ror indexof sizeof adr ref adrinst bitadr add mul div sub trunc move',
+        },
+        contains: [
+            {
+                className: 'string',
+                begin: '\'', end: '\'',
+                contains: [hljs.BACKSLASH_ESCAPE, { begin: '\'\'' }]
```
Comment (pascalfree) [inline src/languages/iecst.js]:
```suggestion
                contains: [hljs.BACKSLASH_ESCAPE, { begin: '\'\'' }],
                relevance: 0
```

File: test/markup/iecst/program.expect.txt [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<span class="hljs-title">CONFIGURATION</span> DefaultCfg
+  <span class="hljs-title">VAR_GLOBAL</span>
+    Start_Stop <span class="hljs-keyword">AT</span> %IX0<span class="hljs-number">.0</span>: <span class="hljs-built_in">BOOL</span>; <span class="hljs-comment">(* This is comment *)</span>
```
Comment (pascalfree) [inline test/markup/iecst/program.expect.txt]: fix markup test
```suggestion
    Start_Stop <span class="hljs-keyword">AT</span> <span class="hljs-symbol">%IX0.0</span>: <span class="hljs-built_in">BOOL</span>; <span class="hljs-comment">(* This is comment *)</span>
```

**Comment by Serhioromano** [comment]
Please close it and delete it. I'll submit a new cleat PR.

**Comment by Serhioromano** [comment]
> On the sample of shell both `iecst` and `shell` score a relevance of 9.
> Improving the relevance should fix this.

Thank you. Although I did not understand how you found this, and why you added relevance for some items and deleted for others. And why relevance in my language file has something to do with other languages at all.

Anyway, it passes checks now and I am happy.

I created new PR #2054 because I am not sure if I could upstream correctly to this very old PR.

**Pull request closed without merging**